### PR TITLE
Replace pylint ruff

### PR DESCRIPTION
**Pull request by marcospri**
This PR:

- Removes existing pylint:disable clause
- Removes pylint configuration
- Removes pylint as a dependency

After merging a version of this we then should:

- Check what more ruff rules we can enable, see https://github.com/hypothesis/lms/pull/6773
- Backport this to the cookiecutter

### CODE REVIEW

File: bin/run_data_task.py [hunk 1]
```diff
@@ -47,7 +47,7 @@
 def main():
     args = parser.parse_args()
 
-    with bootstrap(args.config_file) as env:
+    with bootstrap(args.config_file) as env:  # noqa: PLR1702
```
Comment (marcospri) [inline bin/run_data_task.py:50]: As last time we talked about this is unfornuate that we have to refer to this by code

That's tracked here: https://github.com/astral-sh/ruff/issues/1773


This particular code is for: https://docs.astral.sh/ruff/rules/too-many-nested-blocks/

File: lms/error_code.py [hunk 2]
```diff
@@ -4,10 +4,10 @@
 We have them in their own module to avoid any circular dependency problems.
 """
 
-from enum import Enum
+from enum import Enum, StrEnum
 
 
-class ErrorCode(str, Enum):
+class ErrorCode(StrEnum):
```
Comment (marcospri) [inline lms/error_code.py:10]: From the "UP" type of rules.

File: lms/events/event.py [hunk 3]
```diff
@@ -39,7 +39,7 @@ def serialize(self) -> dict:
             field.name: getattr(self, field.name)
             for field in fields(self)
             # Excluded non-serializable fields
-            if field.name not in ["request", "Type"]
+            if field.name not in {"request", "Type"}
```
Comment (marcospri) [inline lms/events/event.py:42]: Prefer sets for `in` checks.

File: lms/product/factory.py [hunk 4]
```diff
@@ -5,7 +5,6 @@
 def get_product_from_request(request) -> Product:
     """Get the correct product object from the provided request."""
 
-    # pylint:disable=redefined-variable-type
```
Comment (marcospri) [inline lms/product/factory.py:8]: This file is linter/typechecker nightmare.

File: lms/product/product.py [hunk 5]
```diff
@@ -2,7 +2,7 @@
 
 from dataclasses import InitVar, dataclass
 
-from lms.product.family import Family  # pylint:disable=unused-import
+from lms.product.family import Family
```
Comment (marcospri) [inline lms/product/product.py:5]: This import is used below, not sure why we were not getting the pylint alert about it not being needed.

File: lms/scripts/init_db.py [hunk 6]
```diff
@@ -9,7 +9,6 @@
 
 """
 
-# pylint:disable=unused-import
```
Comment (marcospri) [inline lms/scripts/init_db.py:12]: Same here.

File: lms/security.py [hunk 7]
```diff
@@ -95,7 +95,6 @@ def forget(self, request):
     @lru_cache(maxsize=1)
     def get_policy(request: Request):  # noqa: C901, PLR0911
         """Pick the right policy based the request's path."""
-        # pylint:disable=too-many-return-statements
```
Comment (marcospri) [inline lms/security.py:98]: This is PLR0911 above

File: lms/views/admin/_schemas.py [hunk 8]
```diff
@@ -16,7 +16,6 @@ class EmptyStringNoneMixin:
     """
 
     def deserialize(self, value, attr, data, **kwargs):
-        # pylint:disable=compare-to-empty-string
```
Comment (marcospri) [inline lms/views/admin/_schemas.py:19]: This are equivalent, `PLC1901` was disabled nthe config, enabled on  this PR.

File: lms/views/dashboard/api/assignment.py [hunk 9]
```diff
@@ -134,7 +134,6 @@ def assignment(self) -> APIAssignment:
         schema=AssignmentsMetricsSchema,
     )
     def course_assignments_metrics(self) -> APIAssignments:
-        # pylint:disable=too-many-locals
```
Comment (marcospri) [inline lms/views/dashboard/api/assignment.py:137]: While pylint has a point, this method could probably use a refactor adding a more local variables triggers the issue in ruff.

Not sure what the discrepancy in the count is, both linters default to 15 as the threshold.

**Review by seanh — APPROVED** [review]

**Pull request merged**